Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +379,133 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
+		pcpp::Logger::getInstance().enableLogs();
 
-	// try to stop capture from within the callback, verify no impact on capturing
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStopCapture, &packetCount, 10), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+		totalSleepTime = 0;
+		while (totalSleepTime <= 5)
+		{
+			pcpp::multiPlatformSleep(1);
+			totalSleepTime += 1;
+			if (packetCount > 0)
+				break;
+		}
 
-	// verify it's possible to capture non-blocking after the mess done in previous lines
-	packetCount = 0;
-	PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+		PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
 
-	// verify an error returns if trying capture blocking while non-blocking is running
-	pcpp::Logger::getInstance().suppressLogs();
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), 0);
-	pcpp::Logger::getInstance().enableLogs();
+		liveDev->stopCapture();
+		PTF_ASSERT_GREATER_THAN(packetCount, 0);
 
-	totalSleepTime = 0;
-	while (totalSleepTime <= 5)
-	{
-		pcpp::multiPlatformSleep(1);
-		totalSleepTime += 1;
-		if (packetCount > 0)
-			break;
+		liveDev->close();
+		PTF_ASSERT_FALSE(liveDev->isOpened());
+
+		// a negative test
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
+		pcpp::Logger::getInstance().enableLogs();
 	}
+} // TestPcapLiveDeviceBlockingMode
 
-	PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModePollTimeout)
+{
+#if !defined(_WIN32)
+	const char* iptablesAddInputDrop = "sudo iptables -A INPUT -i eth0 -j DROP";
+	const char* iptablesAddOutputDrop = "sudo iptables -A OUTPUT -o eth0 -j DROP";
+	std::system(iptablesAddInputDrop);
+	std::system(iptablesAddOutputDrop);
 
-	liveDev->stopCapture();
-	PTF_ASSERT_GREATER_THAN(packetCount, 0);
+	// open device
+	pcpp::PcapLiveDevice* liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
+	pcpp::PcapLiveDevice::DeviceConfiguration defaultConfig;
+	defaultConfig.usePoll = true;
+
+	PTF_ASSERT_TRUE(liveDev->open(defaultConfig));
+	DeviceTeardown devTeardown(liveDev);
+
+	int packetCount = 0;
+	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, &packetCount, 2), -1);
+	PTF_ASSERT_EQUAL(packetCount, 0);
 
 	liveDev->close();
 	PTF_ASSERT_FALSE(liveDev->isOpened());
 
-	// a negative test
-	pcpp::Logger::getInstance().suppressLogs();
-	PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
-	pcpp::Logger::getInstance().enableLogs();
+	const char* iptablesDeleteInputDrop = "sudo iptables -D INPUT -i eth0 -j DROP";
+	const char* iptablesDeleteOutputDrop = "sudo iptables -D OUTPUT -o eth0 -j DROP";
+	std::system(iptablesDeleteInputDrop);
+	std::system(iptablesDeleteOutputDrop);
```

Comment:
Maybe we can derive from `DeviceTeardown` and add this to it?

Reply:
IMO, the interface and the device are separate concepts. do you still want to do it?

Reply:
Maybe add another teardown class to delete these `iptables` rules?
It's important to always remove them in teardown because if this test fails and we don't remove them, other tests will fail too

Reply:
sure, just added `SystemCommandTeardown`